WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -477,7 +495,7 @@ - (void)deletePostsFromSiteWithID:(NSNumber *)siteID
 {
     NSError *error;
     NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:@"ReaderPost"];
-    request.predicate = [NSPredicate predicateWithFormat:@"siteID = %@ AND isWPCom = YES", siteID];
+    request.predicate = [NSPredicate predicateWithFormat:@"siteID = %@ AND isWPCom = YES AND isSavedForLater == NO", siteID];
```

Comment:
I mention this just to open discussion, I definitely do not consider it a blocker. 

There are two places where we append the `AND isSavedForLater == NO` to a predicate, doing so directly to the predicate format, as a string.

However, it seems to me like we are missing two different concerns (I think the right word for it is "orthogonal" concerns), one being the predicate specific to that particular operation (i.e. deleting posts with site id and link) and another one being the filtering out of posts saved for later.

And the thing is we have APIs to compose predicates, in my opinion, precisely to solve this problem. So, what if we encapsulated the filtering out of posts saved for later in its own predicate, and we encapsulate the creation of that predicate in its own separate method, and then we compose those two predicates by means of an `NSCompoundPredicate`? That way we'd have a more clear separation of concerns, we would be able to change / update what "filtering out posts saved for later" means in terms of predicates at one well encapsulated, isolated place, and we would be able to apply that filter (leaving out posts saved for later) in other places. 

What do you think?

Reply:
I totally agree! I think let's ship this, and clean up the predicates separately :)